### PR TITLE
refactor: remove duplicate code from renaming to workflow

### PR DESCRIPTION
**Pull request by ainouzgali**
### What change does this PR introduce?

As part of the renaming of notification template to workflow(in API) - reuse the current usecases for notification template Instead of duplicate all. 

What was done in this pr:  
- deleted the duplicated workflows folder 
- renamed current notification-templates folder to workflows
- renamed the templates dtos to workflows 
- use 1 module - workflows - with both controllers 
- refactor all existing api tests to use the workflows path

### CODE REVIEW

File: apps/api/src/bootstrap.ts [hunk 1]
```diff
@@ -103,6 +103,7 @@ export async function bootstrap(expressApp?): Promise<INestApplication> {
     .addTag('Integrations')
     .addTag('Layouts')
     .addTag('Workflows')
+    .addTag('Notification Templates')
```
Comment (ainouzgali) [inline apps/api/src/bootstrap.ts:106]: We still want to show the notification templates in our api reference, but all endpoints will have the deprecated tag.

File: apps/api/src/bootstrap.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ import { validateEnv } from './config/env-validator';
 
 import * as packageJson from '../package.json';
 
-const extendedBodySizeRoutes = ['/v1/events', '/v1/notification-templates', '/v1/layouts'];
+const extendedBodySizeRoutes = ['/v1/events', '/v1/notification-templates', '/v1/workflows', '/v1/layouts'];
```
Comment (ainouzgali) [inline apps/api/src/bootstrap.ts:24]: Added the workflows to the exceptions of a large payload

File: apps/api/src/app/workflows/dto/update-workflow-request.dto.ts [hunk 3]
```diff
@@ -37,9 +37,6 @@ export class UpdateWorkflowRequestDto implements ICreateNotificationTemplateDto
   @IsMongoId()
   notificationGroupId: string;
 
-  @ApiPropertyOptional()
-  active?: boolean;
-
```
Comment (ainouzgali) [inline apps/api/src/app/workflows/dto/update-workflow-request.dto.ts:42]: This is an example of why this refactor was needed- a change/fix for the notification templates needed to be duplicated in the workflows folder.

File: apps/api/src/app.module.ts [hunk 4]
```diff
@@ -42,7 +41,7 @@ const modules: Array<Type | DynamicModule | Promise<DynamicModule> | ForwardRefe
   HealthModule,
   EnvironmentsModule,
   ExecutionDetailsModule,
-  NotificationTemplateModule,
+  WorkflowModule,
```
Comment (LetItRock) [inline apps/api/src/app.module.ts:44]: we should probably use the plural names when calling all these modules... wdyt?

**Pull request merged**